libs/genobs/include/genobs/observable.hpp

```diff
@@ -0,0 +1,42 @@
+#ifndef INCLUDE_OBSERVABLES_OBSERVABLE_HPP
+#define INCLUDE_OBSERVABLES_OBSERVABLE_HPP
+
+#include "algorithms.hpp"
+#include "properties.hpp"
+
+namespace GenObs {
+/**
+ * @brief Meta-Observable that returns the product of two
+ *        other observables.
+ *
+ * The operand observables are stored by privately deriving
+ * from them to get the empty case optimization if they do
+ * not have state.
+ *
+ * @tparam Left left operand of the product.
+ * @tparam Right right operand of the product.
+ */
+template <class Left, class Right> struct Product : Left, Right {
+  Product(Left left = {}, Right right = {})
+      : Left(std::move(left)), Right(std::move(right)) {}
+
+  template <class Particle, class Traits = traits<Particle>>
```

Comment:
I think you should add an indirection here, e.g. introduce a new meta function, maybe called `default_traits` that selects the traits class. Maybe

```c++
template<class Particle>
using default_traits = std::decay_t<traits<Particle>>;
```
in the traits header and then

```c++
template <class Particle, class Traits = default_traits<Particle>>
```
here